### DISCUSSION TITLE
Parallel? Cross Apply

### DISCUSSION THREAD

**Discussion by SiPurdy** [MarkMpn/Sql4Cds#521] [General]
What is the expected parallel behaviour with Cross Apply? For example, the following query...
``` sql
select
    contactid
    , fullname
from contact
    cross apply (select * from new_CustomAPI(contact.contactid)) as CustomAPIResult
where
    contact.statecode = 0
```
...seems to run each call to new_CustomAPI sequentially, 1 at a time irrespective of MAXDOP or USE HINT('BATCH_SIZE_?') in the query or the properties on the Query Execution tab within Settings.

**Comment by MarkMpn** [comment]
At present the entire execution plan runs sequentially to build the result set, and parallelism is only used within DML operations. It should be feasible to add parallelism operators in future to enhance this further.
